### DISCUSSION TITLE
How to route GET requests from HTML forms

### DISCUSSION THREAD

**Discussion by alphabitserial** [rwf2/Rocket#2198] [Questions] [answered]
I have an HTML / Bootstrap 5 form that I want to use as a simple search box. I'm using Rocket 0.5.0-rc2.

The HTML looks like this:
```html
<form class="d-flex" action="/posts">
    <input class="form-control me-2" type="text" placeholder="Search by tag..." id="tags" name="tags">
    <button class="btn btn-primary">Search</button>
</form>
```

And submitting it results in a GET request like this: `/posts?tags=hello+world`

I'm not sure how to create a route and function that can use the `hello+world` part of the URL. So far I have this:
```rust
#[get("/posts")]
fn posts() -> Template {
    let context = context! {
        title: "Posts",
        rows: rows()
    };
    Template::render("posts", &context)
}

#[get("/posts?tags=")]
fn empty_search() -> Redirect {
    Redirect::to(uri!("/posts"))
}
```

I tried the following: `#[get("/posts?tags=<query>")]` but I get an error `'segment' contains '<' or '>' but is not a dynamic parameter`

Also, thanks for Rocket. I'm building my first web application with it and it's been tremendous so far. The FileServer handler and built-in template support have been really awesome. After a couple weekends spent reading & hacking, my prototype is almost done already. 😄🚀

**Comment by SergioBenitez** [comment, marked as answer]
See https://rocket.rs/v0.5-rc/guide/requests/#query-strings. Effectively:

```rust
#[get("/posts?<tags>")]
fn search(tags: &str) { .. }
```

**Reply by alphabitserial** [reply]
Oh! I had looked at that, but didn't realize I needed to exclude the "tags?=" bit from the route itself. Thanks so much!
